### PR TITLE
[tests/loggers] refactor with BoringModel

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of #4501

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

File: tests/loggers/test_mlflow.py [hunk 1]
```diff
@@ -148,7 +152,7 @@ def test_mlflow_logger_dirs_creation(tmpdir):
     assert 'epoch' in os.listdir(tmpdir / exp_id / run_id / 'metrics')
     assert set(os.listdir(tmpdir / exp_id / run_id / 'params')) == model.hparams.keys()
     assert trainer.checkpoint_callback.dirpath == (tmpdir / exp_id / run_id / 'checkpoints')
-    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {'epoch=0-step=9.ckpt'}
+    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {'epoch=0-step=63.ckpt'}
```
Comment (awaelchli) [inline tests/loggers/test_mlflow.py:155]: could we set limit_train_batches and put the number here?

File: tests/loggers/test_comet.py [hunk 2]
```diff
@@ -154,12 +154,12 @@ def test_comet_logger_dirs_creation(comet, comet_experiment, tmpdir, monkeypatch
...
     logger.experiment.project_name = 'test'
 
-    model = EvalModelTemplate()
+    model = BoringModel()
     trainer = Trainer(default_root_dir=tmpdir, logger=logger, max_epochs=1, limit_val_batches=3)
     trainer.fit(model)
 
     assert trainer.checkpoint_callback.dirpath == (tmpdir / 'test' / "1" / 'checkpoints')
-    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {'epoch=0-step=9.ckpt'}
+    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {'epoch=0-step=63.ckpt'}
```
Comment (awaelchli) [inline tests/loggers/test_comet.py:162]: see my other comment regarding this step number

**Review by awaelchli — APPROVED** [review]
looking good!

File: tests/loggers/test_mlflow.py (hunk 1 above)
Comment (rohitgr7) [reply tests/loggers/test_mlflow.py:155]: yeah why not.

File: tests/loggers/test_comet.py [hunk 3]
```diff
@@ -154,12 +154,19 @@ def test_comet_logger_dirs_creation(comet, comet_experiment, tmpdir, monkeypatch
...
+        logger=logger,
+        max_epochs=1,
+        limit_train_batches=limit_batches,
+        limit_val_batches=limit_batches,
+    )
     trainer.fit(model)
 
     assert trainer.checkpoint_callback.dirpath == (tmpdir / 'test' / "1" / 'checkpoints')
-    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {'epoch=0-step=9.ckpt'}
+    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {f'epoch=0-step={limit_batches-1}.ckpt'}
```
Comment (carmocca) [inline tests/loggers/test_comet.py:169]:
```suggestion
    assert os.listdir(trainer.checkpoint_callback.dirpath) == [f'epoch=0-step={limit_batches - 1}.ckpt']
```

File: tests/loggers/test_mlflow.py [hunk 4]
```diff
@@ -135,20 +134,27 @@ def test_mlflow_logger_dirs_creation(tmpdir):
...
+        limit_val_batches=limit_batches,
         log_gpu_memory=True,
     )
     trainer.fit(model)
     assert set(os.listdir(tmpdir / exp_id)) == {run_id, 'meta.yaml'}
     assert 'epoch' in os.listdir(tmpdir / exp_id / run_id / 'metrics')
     assert set(os.listdir(tmpdir / exp_id / run_id / 'params')) == model.hparams.keys()
     assert trainer.checkpoint_callback.dirpath == (tmpdir / exp_id / run_id / 'checkpoints')
-    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {'epoch=0-step=9.ckpt'}
+    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {f'epoch=0-step={limit_batches-1}.ckpt'}
```
Comment (carmocca) [inline tests/loggers/test_mlflow.py:157]:
```suggestion
    assert os.listdir(trainer.checkpoint_callback.dirpath) == [f'epoch=0-step={limit_batches - 1}.ckpt']
```

File: tests/loggers/test_wandb.py [hunk 5]
```diff
@@ -146,12 +146,19 @@ def test_wandb_logger_dirs_creation(wandb, tmpdir):
...
+        logger=logger,
+        max_epochs=1,
+        limit_train_batches=limit_batches,
+        limit_val_batches=limit_batches,
+    )
     trainer.fit(model)
 
     assert trainer.checkpoint_callback.dirpath == str(tmpdir / 'project' / version / 'checkpoints')
-    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {'epoch=0-step=9.ckpt'}
+    assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {f'epoch=0-step={limit_batches-1}.ckpt'}
```
Comment (carmocca) [inline tests/loggers/test_wandb.py:161]:
```suggestion
    assert os.listdir(trainer.checkpoint_callback.dirpath) == [f'epoch=0-step={limit_batches - 1}.ckpt']
```

**Review by carmocca — APPROVED** [review]
